src/Cardano/Wallet/Primitive/Types.hs

```diff
@@ -212,6 +220,39 @@ instance Buildable TxOut where
 instance Buildable (TxIn, TxOut) where
     build (txin, txout) = build txin <> " ==> " <> build txout
 
+data TxMeta = TxMeta
+    { metaId :: !(Hash "Tx")
+    -- FIXME: we might want to go for something more optimized than natural,
+    -- like Word64 ?
+    , depth :: !(Quantity "block" Natural)
+    , status :: !TxStatus
+    -- FIXME: should we use Quantity "lovelace" Natural ?
+    -- I wonder could we use somethine like proposed instead of Coin in wallet
+    -- part alltogether. Node/core bits would still work optimally with
+    -- Coin/Word64
+    -- but wallet bits could go a bit more abstract (and less optimized)?
+    , amount :: !Coin
+    , direction :: !Direction
+    , timestamp :: !Timestamp
+    , slotId :: !SlotId
+    } deriving (Show, Eq, Generic)
+
+data TxStatus
+    = Pending
+    | InLedger
+    | Invalidated
+    deriving (Show, Eq, Generic)
+
+data Direction
+    = Outgoing
+    | Incoming
+    deriving (Show, Eq, Generic)
+
+newtype Timestamp = Timestamp
+    -- NOTE: We could have gone for Microsecond from Data.Time.Units, but
+    -- it seems like an overkill for now
+    { getTimestamp :: UTCTime
```

Comment:
in the old wallet we used `Microsecond` which at this point seems like not needed.